evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -205,39 +205,83 @@ def get_prediction_intervals(
         Raises:
             MethodPropertyNotFoundError: If the estimator does not support Time Series Regression as a problem type.
         """
-        X_no_datetime, y_no_datetime = self._drop_time_index(X, y)
-        estimator_input = self.transform_all_but_final(
-            X_no_datetime,
-            y_no_datetime,
-            X_train=X_train,
-            y_train=y_train,
-        )
         has_stl = STLDecomposer.name in list(
             self.component_graph.component_instances.keys(),
         )
         if coverage is None:
             coverage = [0.95]
 
         if self.estimator.model_family in self.NO_PREDS_PI_ESTIMATORS and has_stl:
+            if self.problem_type == ProblemTypes.MULTISERIES_TIME_SERIES_REGRESSION:
+                from evalml.pipelines.utils import stack_data, unstack_multiseries
+
+                X, y = unstack_multiseries(
+                    X,
+                    y,
+                    self.series_id,
+                    self.time_index,
+                    self.input_target_name,
+                )
+
+            X_no_datetime, y_no_datetime = self._drop_time_index(X, y)
+
+            estimator_input = self.transform_all_but_final(
+                X_no_datetime,
+                y_no_datetime,
+                X_train=X_train,
+                y_train=y_train,
+            )
             pred_intervals = self.estimator.get_prediction_intervals(
                 X=estimator_input,
                 y=y,
                 coverage=coverage,
             )
-            trans_pred_intervals = {}
+            intervals_labels = list(list(pred_intervals.values())[0].keys())
+            interval_series_pred_intervals = {
+                interval: {} for interval in intervals_labels
+            }
             residuals = self.estimator.predict(
                 estimator_input,
-            )  # Get residual values
-            trend_pred_intervals = self.get_component(
-                "STL Decomposer",
-            ).get_trend_prediction_intervals(y, coverage=coverage)
-            for key, orig_pi_values in pred_intervals.items():
-                trans_pred_intervals[key] = pd.Series(
-                    (orig_pi_values.values - residuals.values)
-                    + trend_pred_intervals[key].values
-                    + y.values,
-                    index=orig_pi_values.index,
-                )
+            )
+            trans_pred_intervals = {}
+            if self.problem_type == ProblemTypes.MULTISERIES_TIME_SERIES_REGRESSION:
+                trend_pred_intervals = self.get_component(
+                    "STL Decomposer",
+                ).get_trend_prediction_intervals(y_no_datetime, coverage=coverage)
```

Comment:
Why do we need to pass in `y_no_datetime` for multiseries but regular `y` for single series? Is there any way we can get those to match up?